pkg/apis/triggers/v1alpha1/trigger_template_types.go

```diff
@@ -64,3 +65,10 @@ type TriggerTemplateList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []TriggerTemplate `json:"items"`
 }
+// GetApiVersionAndKind returns the apiVersion and Kind for the resourceTemplate
+// Missing fields are represented by empty strings
+func (trt *TriggerResourceTemplate) getApiVersionAndKind() (string, string) {
```

Comment:
Maybe we could use this within the the sink package `createResource()` as well?

Reply:
Yeah, definitely!

Reply:
gonna do this in a follow up. Since the function is on the resourcetemplate and not the triggerTemplate itself, we have to modify a `template.NewResources` as well as `createResource`